### PR TITLE
DOC: Improve README instructions & fixes

### PR DESCRIPTION
**Pull request by viv511**
PR #3 transitioned the instructions into Markdown syntax, but some issues arose from this. 

1. [These build lines](https://github.com/InsightSoftwareConsortium/ITKDoxygen/pull/3/commits/fa2ca68cffb8fe55474d4eeb362b1ebf81c7e610#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R26-R27) are synonymous and should be merged into one

If we look at PR #3, the first line was commented out before, but during the PR it was uncommented leading to this redundancy. The only difference was the build argument passing in the `TAG` variable, but if we examine the [Dockerfile](https://github.com/InsightSoftwareConsortium/ITKDoxygen/blob/master/Dockerfile) there is no mention of this. Thus, the `TAG` variable seems to be unnecessary. 

2. Incorrect grouping of commands before/after the `or`

The location of the `or` and the way the commands were grouped were incorrect. Instead of everything being clumped together, the `or` should have separated the two ways to run the Docker container. Additionally, for those who wish to build a modified version of the documentation to test changes, I removed the [`readonly`](https://github.com/InsightSoftwareConsortium/ITKDoxygen/blob/master/README.md?plain=1#L41) part of the command.

3. Added separation of commands and numbered steps in order to improve readability

Hope this helps! Feedback is appreciated :)
@viv511

### CODE REVIEW

**Review by jhlegarreta — COMMENTED** [review]
Thanks for doing this.

A few comments:

- Why does the `TAG` value that specific value? Can that be explained?
- I would remove `# **IMPORTANT** # !ITK is your absolute path to the modified ITK repository` from the verbatim block, and put after the block explaining the parameters.
- `source=!ITK,` is the exclamation mark necessary? Can that be explained?
- What is the purpose of  copying the doxygen documentation? Can this be explained?
- Add a note explaining the use of the `readonly` parameter when production documentation needs to be generated.

**Comment by viv511** [comment]
Hi @jhlegarreta, 

Thank you so much for your review. I've addressed all of your comments and improved much of the README. 

Best,
@viv511 

Also, I wanted to thank everyone involved with ITK once again for helping me throughout this process. I'm still a high school senior, but I'm excited to continue working on open-source projects like this one. If there are any related issues in the future, I'd always love to help out!

File: README.md [hunk 1]
```diff
@@ -25,23 +24,53 @@ https://insightsoftwareconsortium.github.io/ITKDoxygen/
...
 ```shell
-docker build --build-arg TAG=v1.1.0 . -f Dockerfile -t itk-doxygen
-docker build -f Dockerfile -t itk-doxygen .
+docker build . -f Dockerfile -t itk-doxygen
+```
 
-export TAG=v1.2.0
-docker run --env TAG --name itk-dox itk-doxygen
+### 3. Export TAG Environment Variable
+The `TAG` environment variable is used to specify a filename for the tarballs to be copied from the Docker container to your local working directory (see step 5). Useful for naming multiple versions of the documentation. Here we have set it to *alpha*, but you may change it as needed.
```
Comment (jhlegarreta) [inline README.md:35]: Maybe you can create a cross-reference to step 5 when mentioning it.

Comment (jhlegarreta) [inline README.md:35]: Maybe you can create a cross-reference to step 5 when mentioning it.

File: README.md [hunk 2]
```diff
@@ -25,23 +24,53 @@ https://insightsoftwareconsortium.github.io/ITKDoxygen/
...
+#### Option 2: Using a Local ITK Repository
+If you want to build documentation from a local ITK repository, remove the `readonly` parameter from the mount command and & specify an absolute path to the aforementioned repository.  
 ```shell
 docker run \
   --env TAG \
-  --mount type=bind,source=!ITK,destination=/work/ITK,readonly \
+  --mount type=bind,source=/path/to/your/clone,destination=/work/ITK,readonly \
   --name itk-dox itk-doxygen
+```
+> **Note:** Both of these commands may take a while to complete
```
Comment (jhlegarreta) [inline README.md:54]: Both these commands (...)

File: README.md [hunk 3]
```diff
@@ -25,23 +24,53 @@ https://insightsoftwareconsortium.github.io/ITKDoxygen/
...
 
-or
+### 4. Run the Docker Container
+#### Option 1: Using the Latest ITK Repository
+```shell
+docker run --env TAG --name itk-dox itk-doxygen
+```
 
+#### Option 2: Using a Local ITK Repository
+If you want to build documentation from a local ITK repository, remove the `readonly` parameter from the mount command and & specify an absolute path to the aforementioned repository.
```
Comment (jhlegarreta) [inline README.md:47]: from the mount command and & specify : remove `&`.

File: README.md [hunk 4]
```diff
@@ -1,6 +1,5 @@
 ITKDoxygen
 ==========
-
```
Comment (jhlegarreta) [inline README.md:3]: Unnecessary removal?

**Review by jhlegarreta — COMMENTED** [review]
Thanks for this work @viv511. A couple of additional inline comments.

While addressing those, can you please squash all your commits into a single commit and limit the commit body message line length to 72 characters?

File: README.md (hunk 1 above)
Comment (viv511) [reply README.md]: Sounds good, added!

File: README.md (hunk 4 above)
Comment (viv511) [reply README.md]: Yes, thanks for catching it!

**Comment by viv511** [comment]
Hi! I've made all of the changes that you mentioned. 

I used the `git rebase` command to squash all commits into the "Improve README" commit. However, after rebasing, I realized I had also accidentally pushed a hidden `.DS_Store` file, so I made another commit to remove it. I was running into a lot of issues when I tried to group all of these commits again, which is why you see the additional "Removed hidden .DS_Store file" commit. 

I'm sorry about that, please let me know if this will be okay or if I should change anything further.

Regards,
@viv511

**Comment by jhlegarreta** [comment]
Thanks @viv511. Please squash all commits except 7390d4947981a553f7dc1fec087e219fac1d048d into a single commit explaining the changes. Thanks.

**Comment by viv511** [comment]
Hi!

Thank you so much for your guidance through the commands. Due to an oversight on my part, the `HEAD~7` included a few commits not authored by me. Instead of dropping them as I should have, I left them set to default/`pick`. This eventually caused a lot of issues so I had to reset my local branch to a previous HEAD using `reflog` and `reset` (I understand the caution needed when using `--hard`, but as it was my own fork I made sure to reset at the correct HEAD before any rebasing occurred). When solving merge conflicts in the README file, part of #6 (I suspect it is an issue with VSCode Merge Editor) didn't show up (the build docker image wasn't linked and was rendered as plaintext)  so I had to redo this process again. I did a lot of trial and error but I think I figured out everything that went wrong, so hopefully everything is okay!

tl;dr I think everything is fixed & all of my changes have been rebased into one meaningful commit message. :)

@viv511

File: README.md [hunk 5]
```diff
@@ -1,14 +1,17 @@
...
 
 [doxygen-img]: https://github.com/InsightSoftwareConsortium/ITKDoxygen/actions/workflows/Doxygen.yml/badge.svg
 [doxygen-link]: https://github.com/InsightSoftwareConsortium/ITKDoxygen/actions/workflows/Doxygen.yml
 
 [deployment-img]: https://github.com/InsightSoftwareConsortium/ITKDoxygen/actions/workflows/pages/pages-build-deployment/badge.svg
 [deployment-link]: https://insightsoftwareconsortium.github.io/ITKDoxygen/
 
+[build-docker-img]: https://github.com/InsightSoftwareConsortium/ITKDoxygen/actions/workflows/build-docker-image.yml/badge.svg
+[build-docker-link]: https://github.com/InsightSoftwareConsortium/ITKDoxygen/actions/workflows/build-docker-image.yml
+
```
Comment (jhlegarreta) [inline README.md:14]: Why are all the above changes included in this PR?

These were added in
https://github.com/InsightSoftwareConsortium/ITKDoxygen/pull/6/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5

**Review by jhlegarreta — COMMENTED** [review]
Unfortunately, it does not look like things were cleanly rebased. Probably it will now be easier since you are down to a single commit and figured out things. See the inline comment. Let us know if you need help rebasing.

**Review by blowekamp — APPROVED** [review]
Looks like a nice improvement to the documentation.

Please address @jhlegarreta  comments before merging.

**Comment by viv511** [comment]
> Unfortunately, it does not look like things were cleanly rebased. Probably it will now be easier since you are down to a single commit and figured out things. See the inline comment. Let us know if you need help rebasing.

I believe everything should be corrected and consolidated within [this commit.](https://github.com/InsightSoftwareConsortium/ITKDoxygen/commit/60d249f733e6b7be320b8c662cbf0966b607c821)

Thanks for your guidance & patience,
@viv511

File: README.md (hunk 5 above)
Comment (viv511) [reply README.md]: I believe that I incorrectly added a `drop` in front of that commit instead of removing it from the rebase, which is why the changes in that commit were "dropped". Because those lines were gone, I had to manually add them within the VSCode Merge Editor.

> When solving merge conflicts in the README file, part of https://github.com/InsightSoftwareConsortium/ITKDoxygen/pull/6 (I suspect it is an issue with VSCode Merge Editor) didn't show up

I'm fairly certain the issue didn't lie within VSCode Merge Editor but rather my own faulty rebase.

Comment (jhlegarreta) [reply README.md]: :+1: Resolving.

**Review by jhlegarreta — COMMENTED** [review]
One last comment: although we unfortunately do not use an automatic formatter for our md or rst files, limiting the line length to some reasonable value (e.g. 72, 79, etc.) improves readability by reducing the horizontal scroll. So we need to do that manually, and we are usually not consistent, but can you please do that (choose 72 or 79) and then
```
git commit --amend --no-edit
git push -f origin README-fix
```

please?

Sorry for this long review cycle.

**Comment by viv511** [comment]
No worries!

Do you want me to limit the line length on the original section of the documentation as well? 

Namely, this section:
> This dockerfile has been created to facilitate the creation of [ITK](https://github.com/InsightSoftwareConsortium/ITK) [Doxygen](https://doxygen.nl/) documentation in a reproducible way.
>
> The documentation is built nightly on the current/latest ITK commit, and deployed to GitHub pages at: https://insightsoftwareconsortium.github.io/ITKDoxygen/
>
> To create the the documentation locally, run the following commands:

I was also wondering if I could use the `[]()` link syntax on the aforementioned second line to improve readability.

**Comment by jhlegarreta** [comment]
> Do you want me to limit the line length on the original section of the documentation as well?

Yes, you can do that.

> I was also wondering if I could use the []() link syntax on the aforementioned second line to improve readability.

I think in this case it is preferable to leave the link as is so that the URL can be seen.

**Comment by viv511** [comment]
> No worries!
> 
> Do you want me to limit the line length on the original section of the documentation as well?
> 
> Namely, this section:
> 
> > This dockerfile has been created to facilitate the creation of [ITK](https://github.com/InsightSoftwareConsortium/ITK) [Doxygen](https://doxygen.nl/) documentation in a reproducible way.
> > The documentation is built nightly on the current/latest ITK commit, and deployed to GitHub pages at: https://insightsoftwareconsortium.github.io/ITKDoxygen/
> > To create the the documentation locally, run the following commands:
> 
> I was also wondering if I could use the `[]()` link syntax on the aforementioned second line to improve readability.

Sounds great!

While introducing the line character limits, I found another error with a duplicate "the" on [this line](https://github.com/InsightSoftwareConsortium/ITKDoxygen/blob/master/README.md?plain=1#L26) that was originally there, just wanted to let you know that I'll remove this redundant "the".

File: README.md [hunk 6]
```diff
@@ -17,31 +17,80 @@ Overview
 
 This dockerfile has been created to facilitate the creation of
 [ITK](https://github.com/InsightSoftwareConsortium/ITK)
-[Doxygen](https://doxygen.nl/) documentation in a reproducible way.
+[Doxygen](https://doxygen.nl/)
+
+documentation in a reproducible way.
```
Comment (jhlegarreta) [inline README.md:22]: Why is the above empty line added?

File: README.md [hunk 7]
```diff
@@ -17,31 +17,80 @@ Overview
...
+
+The documentation is built nightly on the current/latest ITK commit,
+
+and deployed to GitHub pages at:
```
Comment (jhlegarreta) [inline README.md:26]: Same comment as above.

**Review by jhlegarreta — CHANGES_REQUESTED** [review]

File: README.md (hunk 6 above)
Comment (viv511) [reply README.md:22]: I used an empty line to limit the character length of each line, but this made each line its paragraph. Instead, I've committed changes that use the `<br>` tags to add line breaks. I separated the section from the original documentation using this 79-character width line as well.

Let me know if the new commit solves this issue!

Comment (viv511) [reply README.md:21]: The solution with `<br>` worked in Markdown preview but didn't render correctly on Github itself. I tried to add a double space at the end of each line and this worked the intended way. 

Sorry for the confusion,
@viv511

Comment (jhlegarreta) [reply README.md:21]: Please, do not add spaces/double spaces at the end of lines; when you hit 79 chars, just start a new line, and that is enough.

**Review by jhlegarreta — CHANGES_REQUESTED** [review]

**Comment by viv511** [comment]
Hi @jhlegarreta, let me know if the changes I pushed last week solve the issue. 

Best,
@viv511

**Review by jhlegarreta — APPROVED** [review]

**Comment by jhlegarreta** [comment]
Thanks @viv511. Merged now.
